assets/js/theme.js

```diff
@@ -49,44 +49,41 @@ let setHighlight = (theme) => {
     document.getElementById("highlight_theme_dark").media = "none";
     document.getElementById("highlight_theme_light").media = "";
   }
-}
-
+};
 
 let setGiscusTheme = (theme) => {
-
   function sendMessage(message) {
-    const iframe = document.querySelector('iframe.giscus-frame');
+    const iframe = document.querySelector("iframe.giscus-frame");
     if (!iframe) return;
-    iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
+    iframe.contentWindow.postMessage({ giscus: message }, "https://giscus.app");
   }
 
   sendMessage({
     setConfig: {
-      theme: theme
-    }
+      theme: theme,
+    },
   });
-
-}
-
+};
 
 let transTheme = () => {
   document.documentElement.classList.add("transition");
   window.setTimeout(() => {
     document.documentElement.classList.remove("transition");
-  }, 500)
-}
-
+  }, 500);
+};
 
 let initTheme = (theme) => {
-  if (theme == null || theme == 'null') {
+  if (theme == null || theme == "null") {
     const userPref = window.matchMedia;
-    if (userPref && userPref('(prefers-color-scheme: dark)').matches) {
-        theme = 'dark';
+    if (userPref && userPref("(prefers-color-scheme: dark)").matches) {
+      theme = "dark";
     }
   }
 
-  setTheme(theme);
-}
-
+  // Wait for the DOM to finish loading before setting theme
+  document.addEventListener("DOMContentLoaded", function () {
+    setTheme(theme);
+  });

```

Comment:
if I understand correctly, this is the main change. everything else is small refactors?

Reply:
actually, this change seems to introduce flickering effect: when you switch to dark mode and then start clicking on other pages, it first loads the new page in the light mode and then switches it to dark mode, creating a weird transition.

Reply:
Yup. The other refactors were proposed by Prettier. Good catch on the flickering effect. Let me see how I can fix it.